types/bytes_amount.go

```diff
@@ -116,6 +116,16 @@ func (z *BytesAmount) Mul(y *BytesAmount) *BytesAmount {
 	return &newZ
```

Comment:
nit & not your bug: The comments for `Mul` `Sub` and `Add` mistakenly reference variable `x` when they actually mean `z`.  Could you fix this as you've fixed the other comments below?

Reply:
Yep indeed. Just fixed them all.